File: 0643-maximum-average-subarray-i/solution.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int n = nums.length;
        int i=0;
        int j=0;
        double sum = 0;
        double maxAvg = Double.NEGATIVE_INFINITY;

        while(j < n) {
            sum += nums[j];

            if(j-i+1 == k) {
                double avg = (double)sum/k;
                maxAvg = Math.max(maxAvg, avg);

                // slide window
                sum -= nums[i];
                i++;
            }
            j++;
        }
        return maxAvg;
    }
}

File: 0643-maximum-average-subarray-i/solution.swift
class Solution {
    func findMaxAverage(_ nums: [Int], _ k: Int) -> Double {
        var i=0;
        var j=0;
        var n = nums.count;
        var sum: Double = 0
        var maxAvg = -Double.infinity

        while j < n {
            sum += Double(nums[j])

            if(j-i+1 == k) {
                maxAvg = max(maxAvg, sum / Double(k))
                sum -= Double(nums[i])
                i += 1
            }
            j += 1
        }
        return maxAvg
    }
}
